src/modules/ekf2/ekf2_main.cpp

```diff
@@ -336,6 +336,10 @@ class Ekf2 final : public ModuleBase<Ekf2>, public ModuleParams
 		(ParamExtFloat<px4::params::EKF2_BARO_NOISE>) _baro_noise,	///< observation noise for barometric height fusion (m)
 		(ParamExtFloat<px4::params::EKF2_BARO_GATE>)
 		_baro_innov_gate,	///< barometric height innovation consistency gate size (STD)
+		(ParamInt<px4::params::EKF2_GND_EFF_EN>)
+		_enableGndEff,	///< Controls barometric deadzone fusion, 0 disables, 1 enables
```

Comment:
@mhkabir  thanks!
@priseborough 

> The major problem with this is that it turns on ground effect compensation permanently when the parameter is set. This is not desirable because it can lead to significant height errors during normal flying. It should only be turned on for takeoff or when landing is expected. This should be done by the commander module.

Agree. But this feature should work in ALT/POS_CTL too. How to define if the landing is expected in these modes?
Thanks!

Reply:
@mhkabir could you please give a hint?

> It should only be turned on for takeoff or when landing is expected.

How to define if the landing is expected in ALT_CTL/POS_CTL modes when only baro is used for height estimation?

> This should be done by the commander module.

I looked into `commander` module but didn't find much about.